st2common/st2common/services/action.py

```diff
@@ -548,3 +551,17 @@ def _is_notify_empty(notify_db):
     if not notify_db:
         return True
     return not (notify_db.on_complete or notify_db.on_success or notify_db.on_failure)
+
+
+def _is_notify_skipped(liveaction):
+    """
+    notification is skipped if action execution is under workflow context and
+    task is not specified under wf_ex_db.notify["tasks"].
+    """
+    if not workflow_service.is_action_execution_under_workflow_context(liveaction):
```

Comment:
I believe this will only work / correctly detect Orquesta workflows, but not action-chain ones, right?

Does this issue / bug only affect Orquesta workflows or also action-chain ones? If it also affects action chain ones, we probably also need to handle that scenario here

Reply:
Verified, issue exists for action-chain workflows as well. Modified the changes to fix same for action-chains. The fix isn't as direct as it for orquesta, as skip_notify parameter is available only in runner parameters. Please review and let me know if the changes look ok to you. Keeping this thread unresolved till then.

Reply:
Thanks for also tackling this for action chain :+1:

Reply:
@Kami If the changes look ok to you, can you please approve and merge? Thanks :)